include/range/v3/utility/iterator.hpp

```diff
@@ -948,12 +933,13 @@ namespace ranges
                 {
                     *it_ = std::move(t);
                 }
-                CONCEPT_REQUIRES(Readable<I>())
-                auto get() const
-                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
-                (
-                    *it_
-                )
+                template<typename II = I,
+                    CONCEPT_REQUIRES_(Same<I, II>() && Readable<II>())>
+                iterator_reference_t<II const> get() const
+                    noexcept(noexcept(*std::declval<II const&>()))
+                {
+                    return *it_;
+                }
```

Comment:
Similar issue with output iterators that do not have `operator*() const` - this can probably be simplified as with `counted_iterator` to:
```c++
CONCEPT_REQUIRES(Readable<I>())
iterator_reference_t<I> get() const
```